default.nix

```diff
@@ -39,92 +45,125 @@ let
-    cacert
-    git
-    findutils
-    patchelf
-  ];
-
+  ] ++ nixpkgs.lib.optional useCustomRocksDb customRocksdb;
+  cargoBuildOptions = opts:
+    packageFlags
+    ++ [featuresFlag]
+    ++ ["--locked" "--offline"]
+    ++ opts;
   shellHook = ''
-    export RUST_SRC_PATH="${rustToolchain}/lib/rustlib/src/rust/src"
-    export LIBCLANG_PATH="${llvm.libclang.lib}/lib"
-    export PROTOC="${protobuf}/bin/protoc"
-    export CFLAGS=" \
-        ${"-isystem ${llvm.libclang.lib}/lib/clang/${llvmVersionString}/include"} \
-        $CFLAGS
-    "
-    export CXXFLAGS+=" \
-        ${"-isystem ${llvm.libclang.lib}/lib/clang/${llvmVersionString}/include"} \
-        $CXXFLAGS
-    "
-    # From: https://github.com/NixOS/nixpkgs/blob/1fab95f5190d087e66a3502481e34e15d62090aa/pkgs/applications/networking/browsers/firefox/common.nix#L247-L253
-    # Set C flags for Rust's bindgen program. Unlike ordinary C
-    # compilation, bindgen does not invoke $CC directly. Instead it
-    # uses LLVM's libclang. To make sure all necessary flags are
-    # included we need to look in a few places.
-    export BINDGEN_EXTRA_CLANG_ARGS=" \
-        ${"-isystem ${llvm.libclang.lib}/lib/clang/${llvmVersionString}/include"} \
-        $BINDGEN_EXTRA_CLANG_ARGS
-    "
-    export ROCKSDB_LIB_DIR="${customRocksdb}/lib"
-    export ROCKSDB_STATIC=1
+    ${rocksDbShellHook}
+    export SUBSTRATE_CLI_GIT_COMMIT_HASH=${SUBSTRATE_CLI_GIT_COMMIT_HASH}
+    export RUSTFLAGS="${rustflags}"
+    export LIBCLANG_PATH=${LIBCLANG_PATH};
+    export PROTOC=${PROTOC}
+    export BINDGEN_EXTRA_CLANG_ARGS="${BINDGEN_EXTRA_CLANG_ARGS}"
+    export CFLAGS="${CFLAGS}"
+    export CXXFLAGS="${CXXFLAGS}"
   '';
-
-  buildPhase = ''
+  preBuild = ''
     ${shellHook}
-    export CARGO_HOME="$out/cargo"
-    export CARGO_BUILD_TARGET="x86_64-unknown-linux-gnu"
-
-    cargo build --locked --release -p aleph-node
   '';
-
-  installPhase = ''
-    mkdir -p $out/bin
-    mv target/x86_64-unknown-linux-gnu/release/aleph-node $out/bin/
+  postInstall = ''
+    if [ -f ${pathToWasm} ]; then
+      mkdir -p $out/lib
+      cp ${pathToWasm} $out/lib/
+    fi
+    if [ -f ${pathToCompactWasm} ]; then
+      mkdir -p $out/lib
+      cp ${pathToCompactWasm} $out/lib/
+    fi
   '';
 
-  fixupPhase = ''
-    find $out -type f -exec patchelf --shrink-rpath '{}' \; -exec strip '{}' \; 2>/dev/null
-  '';
+  SUBSTRATE_CLI_GIT_COMMIT_HASH="${gitCommit}";
+  RUSTFLAGS="${rustflags}";
+  ROCKSDB_STATIC=1;
+  LIBCLANG_PATH="${llvm.libclang.lib}/lib";
+  PROTOC="${protobuf}/bin/protoc";
+  # From: https://github.com/NixOS/nixpkgs/blob/1fab95f5190d087e66a3502481e34e15d62090aa/pkgs/applications/networking/browsers/firefox/common.nix#L247-L253
+  # Set C flags for Rust's bindgen program. Unlike ordinary C
+  # compilation, bindgen does not invoke $CC directly. Instead it
+  # uses LLVM's libclang. To make sure all necessary flags are
+  # included we need to look in a few places.
+  BINDGEN_EXTRA_CLANG_ARGS=" \
```

Comment:
This seems a bit of hacky. Where do we use `bindgen` exactly?

Reply:
e.g. https://github.com/rust-rocksdb/rust-rocksdb/blob/3a46074c36a3f1f1f066ca0c9be1a2efc91563e2/librocksdb-sys/build.rs#L36